leftwm-core/src/handlers/window_handler.rs

```diff
@@ -315,16 +311,8 @@ fn set_relative_floating(window: &mut Window, ws: &Workspace, outer: Xyhw) {
     let xyhw = window.requested.map_or_else(
         || ws.center_halfed(),
         |mut requested| {
-            if ws.xyhw.contains_xyhw(&requested) {
-                requested
-            } else {
-                requested.center_relative(outer, window.border);
-                if ws.xyhw.contains_xyhw(&requested) {
-                    requested
-                } else {
-                    ws.center_halfed()
-                }
-            }
+            requested.center_relative(outer, window.border);
```

Comment:
you might want to check that this is contained within the workspace so the window doesnt get cut off, e.g. if the parent is very narrow and at the edge of the screen the dialog is likely to spawn over the edge. Then if it is cut off then we should centre on the workspace so the whole window is visible. (if that makes sense)

Reply:
I'll close this due to the latest commit.